src/modules/claim/services/claim-verification.service.ts

```diff
@@ -0,0 +1,123 @@
+import { Injectable } from '@nestjs/common';
+import { Logger } from '../../logger/logger.service';
+import { IssuerVerificationService } from './issuer-verification.service';
+import { DIDService } from '../../did/did.service';
+import { RoleCredentialResolver } from '../resolvers/credential.resolver';
+import { ProofVerifier } from '@ew-did-registry/claims';
+import {
+  CredentialResolver,
+  RoleEIP191JWT,
+  isEIP191Jwt,
+} from '@energyweb/vc-verification';
+
+@Injectable()
+export class ClaimVerificationService {
+  credentialResolver: CredentialResolver;
+  constructor(
+    private readonly logger: Logger,
+    private readonly didService: DIDService,
+    private readonly issuerVerificationService: IssuerVerificationService
+  ) {
+    this.logger.setContext(ClaimVerificationService.name);
+    this.credentialResolver = new RoleCredentialResolver(didService);
+  }
+
+  /**
+   * Resolve a credential from storage and verify its proof/signature and its issuer's authority
+   *
+   * @param subjectDID The DID to try to resolve a credential for
+   * @param roleNamesapce The role to try to get a credential for. Should be a full role namespace (for example, "myrole.roles.myorg.auth.ewc")
+   * @return void. Returns boolean indicating if credential is verified. Contains array of error messages if not verified.
+   */
+  public async resolveCredentialAndVerify(
+    subjectDID: string,
+    roleNamespace: string
+  ) {
+    const resolvedCredential = await this.credentialResolver.getCredential(
+      subjectDID,
+      roleNamespace
+    );
+    if (!resolvedCredential) {
```

Comment:
Hi @nichonien quick q for you - I am pretty sure looking at your code that it will always return an EIP first for a resolved credential, but want to make sure - if not I will need to add logic here to also verify a JSON LD credential too, but I think priority is always given to JWT, right? Do you think this is sound logic?

Reply:
It is designed to return both a `VC` and `RoleEIP191Jwt`, the outcome changes based on how the `CredentialResolver` is configured. 

But on a major note, it is mostly relying on DIDDocument for resolution (be it from cached DID Document or directly from blockchain) and thus for now it is just `RoleEIP191Jwt` (as only this type of credential are added to DID Document).

Reply:
If you look at how it is done in `passport-did-auth`  https://github.com/energywebfoundation/passport-did-auth/blob/develop/lib/RoleCredentialResolver.ts

This can return both VC and EIP credentials, but because only EIP credentials are being published to DID Document it will always return EIP type credential (this might / would change in future).

Reply:
Thanks @nichonien. So this PR should also add the method to verify a VC as well, correct? Not just a jwt?

Reply:
Yeah, we can add but I am not sure if it is a must as we are issuing both VC and jwt credentials for a role. This way we will always be sure that verifying only RoleEIP191 credentials will validate all scenarios. @jrhender what are your thoughts ?

Reply:
IMO, just EIP191JWT is enough

Reply:
Great, thanks both!

Reply:
@JGiter do you know the trick to getting tests run locally to behave like tests run in the pipeline? I am running npm run test locally and all my tests are passing, but are failing in pipeline. Just want to recreate locally so I can test it locally.

Reply:
May be it is because DID_REGISTRY_ADDRESS isn't set in CI https://github.com/energywebfoundation/ssi-hub/blob/73d6553a820fb9c7b20ed730118a4d8bc9cf41ce/.github/workflows/build-test.yml#L69, but set in your local .env

Reply:
Thanks @JGiter, do you mean I should change the filename to claims.controller.spec.ts from claims.controller.e2e.spec.ts

Reply:
Yes

Reply:
@JGiter there is [already a claim.controller.spec](https://github.com/energywebfoundation/ssi-hub/blob/develop/src/modules/claim/claim.controller.spec.ts) (which really doesn't test anything). Should I just move everything from claims.controller.e2e.spec.ts to this test file?

Reply:
Removed e2e and now just claim.controller.spec